sim/core/apl_values_dot.go

```diff
@@ -222,10 +239,71 @@ func (value *APLValueDotPercentIncrease) GetFloat(sim *Simulation) float64 {
 	if expectedDamage == 0 {
 		return 1
 	}
-
 	return value.spell.ExpectedTickDamage(sim, value.target)/expectedDamage - 1
 }
 
-func (value *APLValueDotPercentIncrease) String() string {
-	return fmt.Sprintf("Dot Percent Increase (%s)", value.spell.ActionID)
+type APLValueDotCritPercentIncrease struct {
+	*APLValueDotIncreaseCheck
+}
+
+func (rot *APLRotation) newValueDotCritPercentIncrease(config *proto.APLValueDotPercentIncrease, _ *proto.UUID) APLValue {
+	parentImpl := rot.newDotIncreaseValue("Dot Crit Chance Percent Increase", config.SpellId, config.TargetUnit)
+	if parentImpl == nil {
+		return nil
+	}
+	return &APLValueDotCritPercentIncrease{APLValueDotIncreaseCheck: parentImpl}
+}
+
+func (value *APLValueDotCritPercentIncrease) Type() proto.APLValueType {
+	return proto.APLValueType_ValueTypeFloat
+}
```

Comment:
The `Type()` method can live in the parent struct to avoid copy-pasting, since the type is the same for all three children.